app/stores/WalletsStore.js

```diff
@@ -194,6 +220,14 @@ export default class WalletsStore extends Store {
       if (!hasActiveWallet && hasAnyWalletsLoaded) this.active = this.all[0];
       if (this.active) this.goToWalletRoute(this.active.id);
     }
-  }
+  };
+
+  _patchWalletRequestWithWallet = async (wallet) => {
+    await this.walletsRequest.patch(result => {
+      const foundWallet = _.find(result, { id: wallet.id });
+      if (foundWallet) result.pop(foundWallet);
+      result.push(wallet);
+    });
+  };
```

Comment:
When we import new wallet is returned along with the others from the `getWallets`. And we are polling that endpoint. Client API is returning imported wallet from `importKey` as a convenience so we can redirect to it.